presto-orc/src/main/java/com/facebook/presto/orc/FilterFunction.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.relation.Predicate;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class FilterFunction
```

Comment:
Do you think this could be a private class inside `OrcSelectiveRecordReader`?  If `deterministic` isn't used, then everything can be determined from the `predicate`.

Reply:
@tdcmeehan Technically, I think it would be possible, but the predicate is defined in terms of column names, hence, we'll need to pass in column names into `OrcSelectiveRecordReader`. I think this would make it harder to reason about that class. I would also like to avoid having logic that operates on expressions there.